src/components/activity/ActivityController.cpp

```diff
@@ -0,0 +1,34 @@
+#include "components/activity/ActivityController.h"
+
+using namespace Pinetime::Controllers;
+
+ActivityController::ActivityController(Controllers::DateTime& dateTimeController, Controllers::Settings& settingsController)
+  : dateTimeController {dateTimeController}, settingsController {settingsController} {
+}
+
+bool ActivityController::UpdateSteps(uint32_t nbSteps) {
+  int8_t utcMinutes = (dateTimeController.Minutes() + dateTimeController.UtcOffset() * 15) % 60;
+  if (utcMinutes < 0) {
+    utcMinutes += 60;
+  }
```

Comment:
Why does this class need to handle UTC? Shouldn't that happen automatically in DateTimeController?

Reply:
DateTimeController's Minutes() function gives it in the local timezone, but for this, we want it in UTC, so that if the user changes to a fractional (i.e. not whole hour difference) timezone, it doesn't mess with the steps buffer.